sdk/eventhubs/azure-messaging-eventhubs-perf/src/main/java/com/azure/messaging/eventhubs/perf/SendEventBatchTest.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs.perf;
+
+import com.azure.messaging.eventhubs.EventData;
+import com.azure.messaging.eventhubs.EventDataBatch;
+import com.azure.messaging.eventhubs.models.CreateBatchOptions;
+import com.azure.messaging.eventhubs.perf.core.EventHubsPerfStressOptions;
+import com.azure.messaging.eventhubs.perf.core.ServiceTest;
+import reactor.core.publisher.Mono;
+
+/**
+ * Runs the Send Events Batch Performance Test for EventHubs.
+ */
+public class SendEventBatchTest extends ServiceTest<EventHubsPerfStressOptions> {
+    private final EventDataBatch eventDataBatch;
+
+    /**
+     * Runs the Send Events in Batch peformance test.
+     *
+     * @param options The options bag to use for testing.
+     * @throws IllegalStateException when the specified number of messages cannot fit in the specified batch size.
+     */
+    public SendEventBatchTest(EventHubsPerfStressOptions options) throws IllegalStateException {
+        super(options);
+        CreateBatchOptions createBatchOptions = getBatchOptions(options);
+        if (createBatchOptions != null) {
+            eventDataBatch = eventHubProducerClient.createBatch(createBatchOptions);
```

Comment:
If this code is making a network request, the `setupAsync()` method would be a better place than the constructor.  The setup and cleanup methods are async-only, and it shouldn't impact throughput if you use the async client in setup/cleanup and the sync client in `run()` (even if this means an extra conection is created), since we only measure the perf of the `run()` method.  I suppose it would slightly increase other perf metrics like memory and number of TCP connections, but it seems minimal.  If you really want to avoid this, then yes you can use the sync client even in `setupAsync()`.  The framework doesn't care if `setupAsync()` actually blocks instead of using async.

In .NET we only use the async APIs for setup/cleanup, though I don't think this creates an extra TCP connection since .NET sync is just sync-over-async.

If we do think this is really important for Java, we should probably modify the base PerfStressTest to have both `setup()/setupAsync()` and `cleanup()/cleanupAsync()`, to make it more clear that tests should be consistent across setup/run/cleanup.